### PR TITLE
Multi thread issue fix

### PR DESCRIPTION
**Pull request by susrisha**
- Fixes the issue with multi threaded approach
- There is an inherent issue in the Library where the files are extracted to the same folder everytime. 
- This is fixed by using 1 concurrent message at a time and processing it
- Also removed creation of a separate thread each time. The messages are received in a threadpool execution manner and need not be handled by a separate thread each time.
### Testing
- This is tested with a sample topic and a sample message is run 10 times and 15 times in two consecutive sessions. Both times, it was able to execute and respond with all the required parameters and details.

### CODE REVIEW

File: src/gtfx_flex_validator.py [hunk 1]
```diff
@@ -18,13 +19,15 @@ class GTFSFlexValidator:
...
-        settings = Settings()
-        self._subscription_name = settings.request_subscription
-        self.request_topic = core.get_topic(topic_name=settings.request_topic_name)
-        self.response_topic = core.get_topic(topic_name=settings.response_topic_name)
-        self.logger = core.get_logger()
-        self.storage_client = core.get_storage_client()
+        self.core = Core()
+        self.settings = Settings()
+        self._subscription_name = self.settings.request_subscription
+        self.request_topic = self.core.get_topic(topic_name=self.settings.request_topic_name,max_concurrent_messages=1)
```
Comment (susrisha) [inline src/gtfx_flex_validator.py:25]: make it env configurable. if not available, pick as 1.

File: src/gtfx_flex_validator.py [hunk 2]
```diff
@@ -34,9 +37,10 @@ def process(message) -> None:
                 gtfs_upload_message = QueueMessage.to_dict(message)
                 upload_msg = FileUploadMsg.from_dict(gtfs_upload_message)
                 logger.info(upload_msg)
-                # upload_message = GTFSFlexUpload.data_from(gtfs_upload_message)
-                process_thread = threading.Thread(target=self.process_message,args=[upload_msg])
-                process_thread.start()
+                # process_thread = threading.Thread(target=self.process_message,args=[upload_msg])
+                # process_thread.start()
+                # process_thread.join()
+                self.process_message(upload_msg)
```
Comment (susrisha) [inline src/gtfx_flex_validator.py:43]: put a try catch here

File: src/config.py [hunk 3]
```diff
@@ -12,6 +12,7 @@ class Settings(BaseSettings):
     response_topic_name: str = os.environ.get('RESPONSE_TOPIC', None)
     request_subscription: str = os.environ.get('REQUEST_SUBSCRIPTION', None)
     storage_container_name: str = os.environ.get('CONTAINER_NAME', 'gtfsflex')
+    max_concurrent_messages: int = os.environ.get('MAX_CONCURRENT_MESSAGES', 1)
```
Comment (MashB) [inline src/config.py:15]: can you please update the readme for MAX_CONCURRENT_MESSAGES variable

File: src/gtfx_flex_validator.py [hunk 4]
```diff
@@ -34,9 +35,7 @@ def process(message) -> None:
                 gtfs_upload_message = QueueMessage.to_dict(message)
                 upload_msg = FileUploadMsg.from_dict(gtfs_upload_message)
                 logger.info(upload_msg)
-                # upload_message = GTFSFlexUpload.data_from(gtfs_upload_message)
-                process_thread = threading.Thread(target=self.process_message,args=[upload_msg])
-                process_thread.start()
+                self.process_message(upload_msg)
```
Comment (MashB) [inline src/gtfx_flex_validator.py:38]: Can we add try catch block to process_message method. In case failure we need to respond to queue about failure

**Review by MashB — CHANGES_REQUESTED** [review]

**Review by MashB — APPROVED** [review]

**Pull request merged**